packit/distgit.py

```diff
@@ -87,16 +87,30 @@ def create_branch(self, branch_nane: str, base: str = "HEAD"):
         # what if the branch already exists?
         self.local_project.git_repo.create_head(branch_nane, commit=base)
 
-    def checkout_branch(self, git_ref: str):
+    def checkout_branch(self, git_ref: str, fetch_latest: bool = False):
         """
         Perform a `git checkout`
+
+        :param git_ref: ref to check out
+        :param fetch_latest: fetch latest content from the remote and check it out
         """
+        origin = self.local_project.git_repo.remote("origin")
+        if fetch_latest:
+            origin.fetch()
+        try:
+            remote_ref = getattr(origin.refs, git_ref)
```

Comment:
What about basic `if git_ref in origin.refs`?

Reply:
I also got an error in tests.

Reply:
ehm, so it also implements the dict interface? I thought it's an attribute.

Reply:
It's `IterableList`, or something like that.

Reply:
yes, it behaves like a list -_-